packages/core/src/animation/Animator.ts

```diff
@@ -223,24 +227,48 @@ export class Animator extends Component {
       animatorStateData = new AnimatorStateData();
       animatorStateDataCollection[stateName] = animatorStateData;
       this._saveAnimatorStateData(animatorState, animatorStateData);
+      this._saveAnimatorEventHandlers(animatorState, animatorStateData);
     }
     return animatorStateData;
   }
 
   private _saveAnimatorStateData(animatorState: AnimatorState, animatorStateData: AnimatorStateData): void {
-    const { entity, _animationCureOwners: animationCureOwners } = this;
-    const { owners } = animatorStateData;
+    const { entity, _animationCurveOwners: animationCureOwners } = this;
+    const { curveOwners } = animatorStateData;
     const { _curves: curves } = animatorState.clip;
     for (let i = curves.length - 1; i >= 0; i--) {
       const curve = curves[i];
       const targetEntity = entity.findByPath(curve.relativePath);
       const { property } = curve;
       const { instanceId } = targetEntity;
       const propertyOwners = animationCureOwners[instanceId] || (animationCureOwners[instanceId] = []);
-      owners[i] =
+      curveOwners[i] =
         propertyOwners[property] ||
-        (propertyOwners[property] = new AnimationCureOwner(targetEntity, curve.type, property));
+        (propertyOwners[property] = new AnimationCurveOwner(targetEntity, curve.type, property));
+    }
+  }
+
+  private _saveAnimatorEventHandlers(state: AnimatorState, animatorStateData: AnimatorStateData) {
+    const { _entityScripts: scripts } = this;
+    const { events } = state.clip;
+    const { eventHandlers } = animatorStateData;
+    scripts.length = 0;
+    eventHandlers.length = 0;
+    this._entity.getComponents(Script, scripts);
+    for (let i = events.length - 1; i >= 0; i--) {
+      const event = events[i];
+      const funcName = event.functionName;
+      for (let j = scripts.length - 1; j >= 0; j--) {
+        const handler = scripts[j][funcName];
+        if (handler) {
+          const eventHandler = new AnimationEventHandler();
+          eventHandler.event = event;
+          eventHandler.handlers.push(handler);
+          eventHandlers.push(eventHandler);
+        }
+      }
     }
+    eventHandlers.sort((a, b) => b.event.time - a.event.time);
```

Comment:
Sort here is not good! You can sort `AnimatorEvent`  when it is added to `Animationclip`!

Reply:
done